core/src/main/scala/neotypes/Async.scala

```diff
@@ -64,13 +73,26 @@ object Async {
           }
         }
 
+      override final def makeLock: Future[Lock] =
+        Future.successful {
+          val q = new ArrayBlockingQueue[Unit](1)
```

Comment:
As always, the future implementation was the most difficult one.

It literary took me 15 minutes to implement all the other effects, whereas this one took me like 5 hours.

Between all the things I tried, **ReentrantLocks** were the first _(that is why I named it `Lock`)_.
They won't work because the EC may schedule two futures in the same thread and it will fail _(it happened multiple times when running tests)_.

Reply:
Is there any concern about not having lock fairness?